### PR TITLE
added tests for page index in parquet files

### PR DESCRIPTION
**Pull request by max-vostrikov**
added tests for https://github.com/ClickHouse/ClickHouse/pull/65475
Support writting page index into parquet file

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [x]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [x]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 928d5df3f02559fab4fd9bbd258f6e9c6ac4c18a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67596/928d5df3f02559fab4fd9bbd258f6e9c6ac4c18a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Push to Dockerhub</td><td>The check for building and pushing the CI related docker images to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67596/928d5df3f02559fab4fd9bbd258f6e9c6ac4c18a/push_to_dockerhub_amd64.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/integration/test_parquet_page_index/test.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import pytest
+from helpers.cluster import ClickHouseCluster
+import pyarrow.parquet as pq
+import os
+import time
+
+cluster = ClickHouseCluster(__file__)
+path_to_userfiles = "/var/lib/clickhouse/user_files/"
+path_to_external_dirs = (
+    "/ClickHouse/tests/integration/test_parquet_page_index/_instances"
```
Comment (serxa) [inline tests/integration/test_parquet_page_index/test.py:10]: I believe there was a better way to obtain path to `_instances` dir. Something like `cluster.instances_dir_name`. But anyway, I don't understand why is it needed. `external_dirs` argument in `cluster.add_instance` function maps this path as absolute in container, but makes it relative to `_instances` dir on host

File: tests/integration/test_parquet_page_index/test.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+def delete_if_exists(file_path):
+    if os.path.exists(file_path):
+        os.remove(file_path)
+
+
+@pytest.mark.parametrize(
+    "query, expected_result",
+    {
+        (
+            "SElECT number, number+1 FROM system.numbers LIMIT 100 "
```
Comment (serxa) [inline tests/integration/test_parquet_page_index/test.py:41]: Minor. There is a small `l` in `SElECT` and in a few places below.

File: tests/integration/test_parquet_page_index/test.py (hunk 1 above)
Comment (max-vostrikov) [reply tests/integration/test_parquet_page_index/test.py]: yes, cluster.instances_dir.
I need this to open the file from `/var/lib/clickhouse/user_files/` later. It is easer to open it on the test host rather then on cluster node, so it is easier to mount `/var/lib/clickhouse/user_files/`

**Review by serxa — APPROVED** [review]

**Pull request merged**